.github/workflows/wheels.yml

```
@@ -1,7 +1,5 @@
 # Generated by maturin v0.14.14 and modified by update-wheels-ci.sh
-on:
-  release:
-    types: [published]
+on: push
```

You can also add an event, without removing the previous one>
```suggestion
on:
  push:
  release:
    types: [published]
```

---

That's going in the right direction; I think I want to run every job except `release` in `wheels.yml` whenever I want to test them (in branches like this one).

---

This is also what Stefano C is doing in quantum repos (even if they are just pure Python... here I believe it's more relevant).

You can filter the jobs and individual steps with `if:` key, as you are doing, and the condition can contain an expression, that can depend on variables in contexts (thus including events and outputs of previous steps).
https://docs.github.com/en/actions/learn-github-actions/expressions
https://docs.github.com/en/actions/learn-github-actions/contexts
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif